common/src/services/api.service.ts

```diff
@@ -2246,7 +2246,8 @@ export class ApiService implements ApiServiceAbstraction {
     );
 
     if (response.status === 200) {
-      return true;
+      const body = await response.json();
+      return body.token;
```

Comment:
This is bypassing type checking because `body.token` has `any` type, but the method signature says to expect a `boolean`. Please keep types in line with expectations.